File: Python/EntradaAleatoria/6_SumN/Sequential/sum_n.py
#!/usr/bin/env python3
"""
SumN sequential - EntradaAleatoria (Python)
Reads n from params.txt if present, otherwise uses a default; prints numbers 1..n and their sum and CPU time.
"""
import time
import sys
import random

DEFAULT_N = 10

def read_n(path='params.txt'):
    try:
        with open(path,'r') as f:
            data = f.read().strip().split()
            if data:
                return int(data[0])
    except Exception:
        return None
    return None

if __name__ == '__main__':
    n = read_n() or DEFAULT_N
    print('--------- SumN&Sequential (EntradaAleatoria - Python) --------')
    start = time.process_time()
    s = 0
    print('\nLos primeros', n, 'numeros naturales son:')
    for i in range(1, n+1):
        s += i
        print(i, end=' ')
    end = time.process_time()
    print('\n\nLa suma es:', s)
    print(f'Tiempo total de ejecución (CPU): {end - start:.6f} segundos')
    print('---------- Program End ----------')

File: 6_SumN/Sequential/SumN.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    int n;
    int sum = 0;

    printf("--------- SumN&Sequential --------\n");
    printf("Ingrese la cantidad de los primeros numeros naturales a sumar: ");
    scanf("%d", &n);

    clock_t start = clock();

    printf("\nLos primeros %d numeros naturales son:\n", n);
    for (int i = 1; i <= n; i++) {
        sum += i;
        printf("%d ", i);
    }

    clock_t end = clock();

    printf("\n\nLa suma de los %d primeros numeros naturales es: %d\n", n, sum);
    printf("Tiempo total de ejecución: %.6f segundos\n",
           (double)(end - start) / CLOCKS_PER_SEC);

    printf("---------- Program End ----------\n");
    return 0;
}
